Content.Server/GameObjects/Components/Construction/ConstructionComponent.cs

```diff
@@ -141,52 +142,13 @@ bool TryProcessStep(ConstructionStep step, IEntity slapped)
                     return true;
                 case ConstructionStepTool toolStep:
-                    switch (toolStep.Tool)
-                    {
-                        case ToolType.Crowbar:
-                            if (slapped.HasComponent<CrowbarComponent>())
-                            {
-                                sound.Play("/Audio/items/crowbar.ogg", Transform.GridPosition);
-                                return true;
-                            }
-                            return false;
-                        case ToolType.Welder:
-                            if (slapped.TryGetComponent(out WelderComponent welder) && welder.TryUse(toolStep.Amount))
-                            {
-                                if (_random.NextDouble() > 0.5)
-                                    sound.Play("/Audio/items/welder.ogg", Transform.GridPosition);
-                                else
-                                    sound.Play("/Audio/items/welder2.ogg", Transform.GridPosition);
-                                return true;
-                            }
-                            return false;
-                        case ToolType.Wrench:
-                            if (slapped.HasComponent<WrenchComponent>())
-                            {
-                                sound.Play("/Audio/items/ratchet.ogg", Transform.GridPosition);
-                                return true;
-                            }
-                            return false;
-                        case ToolType.Screwdriver:
-                            if (slapped.HasComponent<ScrewdriverComponent>())
-                            {
-                                if (_random.NextDouble() > 0.5)
-                                    sound.Play("/Audio/items/screwdriver.ogg", Transform.GridPosition);
-                                else
-                                    sound.Play("/Audio/items/screwdriver2.ogg", Transform.GridPosition);
-                                return true;
-                            }
-                            return false;
-                        case ToolType.Wirecutters:
-                            if (slapped.HasComponent<WirecutterComponent>())
-                            {
-                                sound.Play("/Audio/items/wirecutter.ogg", Transform.GridPosition);
-                                return true;
-                            }
-                            return false;
-                        default:
-                            throw new NotImplementedException();
-                    }
+                    if (!slapped.TryGetComponent<ToolComponent>(out var tool))
+                        return false;
+                    if (toolStep.Tool != tool.Behavior) return false;
+                    if (toolStep.Tool == Tool.Welder && !((WelderComponent)tool).TryWeld(toolStep.Amount)) return false;
```

Comment:
So the problem here is that the powered hand drill you added now can't actually be, well, powered, because you hardcoded it so welders are special cased.

Reply:
I changed it so a virtual UseTool is called on the ToolComponent, making it possible for actual powered tools to exist.
I called Welder's own overload here due to toolsteps specifying the amount of fuel needed.